doc/source/whatsnew/v0.23.1.txt

```diff
@@ -83,6 +83,7 @@ Indexing
 I/O
 ^^^
 
+- Bug in :class:`pandas.io.common.BytesZipFile` where zip compression produces uncompressed zip archive (:issue:`17778`)
```

Comment:
can you reference 21144 (as well is fine) as that other issue was closed for 0.23.0

Reply:
Hi @jreback @WillAyd , tested 21144 on older version of pandas 0.20 on windows and same issue occurred. I think it's an existing issue unrelated to this PR. This PR only address zip compression and doesn't touch gzip at all.